### PR TITLE
Find path to FreeCAD.so when importing from external python on MacOS

### PR DESCRIPTION
**Pull request by ianrrees**
- [x] snuck in before checklists

### CODE REVIEW

File: src/App/Application.cpp [hunk 1]
```diff
@@ -2281,24 +2281,58 @@ std::string Application::FindHomePath(const char* sCall)
...
-        free(buf);
-
-        if (path) {
-            std::string Call(resolved), TempHomePath;
-            std::string::size_type pos = Call.find_last_of(PATHSEP);
-            TempHomePath.assign(Call,0,pos);
-            pos = TempHomePath.find_last_of(PATHSEP);
-            TempHomePath.assign(TempHomePath,0,pos+1);
-            return TempHomePath;
+                if ( !PyString_Check(pyPath) )
```
Comment (ianrrees) [inline src/App/Application.cpp]: I'll add a conditional Python3 version of this shortly

File: src/App/Application.cpp [hunk 2]
```diff
@@ -2281,27 +2281,64 @@ std::string Application::FindHomePath(const char* sCall)
...
+
+#if PY_MAJOR_VERSION >= 3
+                if ( !PyUnicode_Check(pyPath) )
+                    continue;
+                std::string path( PyUnicode_AsUTF8(pyPath) );
+#else 
+                if ( !PyString_Check(pyPath) )
+                    continue;
+                std::string path( PyString_AsString(pyPath) );
+#endif // #if/else PY_MAJOR_VERSION >= 3
```
Comment (looooo) [inline src/App/Application.cpp]: maybe checking for PyUnicode (python2 and 3)? and elif checking for PyString for python2?
````
    if ( !PyUnicode_Check(pyPath) )
        std::string path( PyUnicode_AsUTF8(pyPath) );
#if PY_MAJOR_VERSION < 3
    if ( !PyString_Check(pyPath)
        std::string path( PyString_AsString(pyPath) );
+#endif // #if/else PY_MAJOR_VERSION < 3

Comment (ianrrees) [inline src/App/Application.cpp]: Sure, I'll change it to test for (unicode || string) in Python2 and just unicode in Python3.

However, I should note that this isn't going to add additional functionality until 2973 is fixed.

**Comment by wwmayer** [comment]
Out of curiosity, shouldn't the code to check for FreeCAD.so be located in MainPy.cpp?

**Comment by ianrrees** [comment]
Aha, yes, I think it should.  I'll try to change this PR tomorrow.  Thanks!

**Comment by ianrrees** [comment]
Forgot to comment yesterday that I've updated this to put most of the changes in MainPy.cpp.  Thanks!

**Comment by wwmayer** [comment]
Merged.

**Pull request closed without merging**